From 8456a39482e63996695a325a6312014a050b254d Mon Sep 17 00:00:00 2001
From: ruki <waruqi@gmail.com>
Date: Thu, 13 Jan 2022 22:36:06 +0800
Subject: [PATCH 1/4] add riscv arch

---
 src/tbox/prefix/arch.h | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/tbox/prefix/arch.h b/src/tbox/prefix/arch.h
index feb8a98e..a421972c 100644
--- a/src/tbox/prefix/arch.h
+++ b/src/tbox/prefix/arch.h
@@ -177,7 +177,7 @@
 #       endif
 #   endif
 #elif defined(__loongarch__)
-#   define TB_ARCH_LOONGARCH 
+#   define TB_ARCH_LOONGARCH
 #   if defined(__loongarch64)
 #       define TB_ARCH_STRING               "loongarch64"
 #   elif defined(__loongarch32)
@@ -185,6 +185,15 @@
 #   else
 #       error unknown version of LoongArch, please feedback to us.
 #   endif
+#elif defined(__riscv)
+#   define TB_ARCH_RISCV
+#   if defined(__riscv_xlen) && __riscv_xlen == 64
+#       define TB_ARCH_STRING               "riscv64"
+#   elif defined(__loongarch32)
+#       define TB_ARCH_STRING               "riscv32"
+#   else
+#       error unknown version of RiscV, please feedback to us.
+#   endif
 #elif defined(TB_COMPILER_IS_TINYC)
 #   if defined(TCC_TARGET_I386)
 #       define TB_ARCH_x86
-- 
2.35.1

